crates/burn-tensor/src/tensor/api/check.rs

```diff
@@ -568,10 +568,13 @@ impl TensorCheck {
         check
     }
 
-    pub(crate) fn matmul<B: Backend, const D: usize>(
-        lhs: &Tensor<B, D>,
-        rhs: &Tensor<B, D>,
-    ) -> Self {
+    pub(crate) fn matmul<B: Backend, const D: usize, K>(
+        lhs: &Tensor<B, D, K>,
+        rhs: &Tensor<B, D, K>,
+    ) -> Self
+    where
+        K: BasicOps<B>,
```

Comment:
I would put matmul as a numeric op, not a basic op

Reply:
Done